include/solver.h

```diff
@@ -352,6 +352,9 @@ class AbsSmtSolver
   virtual Term substitute(const Term term,
                           const UnorderedTermMap & substitution_map) const;
 
+  virtual TermVec substitute_terms(
```

Comment:
What's the motivation to add this to `AbsSolver`? Do most other solvers support this natively? Is this expected to be commonly used? If there isn't a good motivation, I wouldn't include this.

Reply:
There's actually a pretty good case for this even if the solver doesn't support it. If you're using the same substitution map for replacements in several terms, it's expensive to copy and unpack it each time (for model checking we could have thousands of elements in the map). If the solver backend _does_ support it, then you also get the benefit of a shared cache.